db/view/view_update_generator.cc

```diff
@@ -97,6 +97,7 @@ view_update_generator::view_update_generator(replica::database& db, sharded<serv
     setup_metrics();
     discover_staging_sstables();
     (void)_proxy;
+    _db.plug_view_update_generator(*this);
```

Comment:
This is safe because storage_proxy::init_messaging_service is called very late. But what about commitlog replay? It's called before. Can it trigger view updates?

I'd be happier if everything related to dependencies were in main() so it could be analyzed.

Reply:
It's sort of in main already -- view update generator starts in main and gets its "dependencies" via sharded<>::start() arguments

Reply:
> But what about commitlog replay? It's called before. Can it trigger view updates?

As far as I see -- commitlog replaying calls `database::apply_in_memory` which doesn't involve pushing view updates. The v.u.g. is needed one level above it -- in `database::do_apply()` that pushes view updates, creates commitlog entry and then calls the `apply_in_memory`